koku/sources/sources_error_message.py

```diff
@@ -0,0 +1,90 @@
+#
+# Copyright 2020 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Sources Error Message."""
+import logging
+
+from rest_framework.serializers import ValidationError
+
+from providers.provider_errors import ProviderErrors
+
+LOG = logging.getLogger(__name__)
+
+
+class SourcesErrorMessage:
+    """Sources Error Message for Sources API service."""
+
+    def __init__(self, error):
+        """Initialize the message generator."""
+        self._error = error
+
+    def azure_client_errors(self, message):
+        """Azure client error messages."""
+        scrubbed_message = ProviderErrors.AZURE_GENERAL_CLIENT_ERROR_MESSAGE
+        if "http error: 401" in message:
```

Comment:
Yes, we should figure out if there are specific Azure exceptions that are thrown for all of these rather than the broad `AzureException` so we don't have to do this error-prone string examining.  I'm looking into it.